### PR TITLE
Support adding custom configurations in `/etc/phpmyadmin/conf.d`

### PR DESCRIPTION
**Pull request by pigochu**
#319

### CODE REVIEW

**Comment by ibennetch** [comment]
Good idea

File: config.inc.php [hunk 1]
```diff
@@ -159,3 +159,9 @@
 if (file_exists('/etc/phpmyadmin/config.user.inc.php')) {
     include '/etc/phpmyadmin/config.user.inc.php';
 }
+
+/* Support additional configurations */
```
Comment (williamdes) [inline config.inc.php:163]: your should check that the folder exists right first

**Review by williamdes — APPROVED** [review]
Awesome, can you add some documentation on the README for that?

**Comment by williamdes** [comment]
You may be interested to translate phpMyAdmin in your language: https://hosted.weblate.org/projects/phpmyadmin/5-2/th/

Did I get it right?

**Comment by pigochu** [comment]
> You may be interested to translate phpMyAdmin in your language: https://hosted.weblate.org/projects/phpmyadmin/5-2/th/
> 
> Did I get it right?
I am from Taiwan.
My English is not good.

**Comment by pigochu** [comment]
> Awesome, can you add some documentation on the README for that?
@williamdes 

I use google translate to translate my thoughts into English, what do you think?
------------------------------------------------------------------------------------------

## Adding Custom Configuration

Instead of using `/etc/phpmyadmin/config.user.inc.php`, you can also consider storing your custom configuration files in /etc/phpmyadmin/conf.d, which is very suitable for managing multiple MySQL Container uses different file names to distinguish settings.

On the `docker run` line like this:

```sh
docker run --name phpmyadmin -d --link mysql_db_server:db -p 8080:80 -v /some/local/directory/conf.d:/etc/phpmyadmin/conf.d phpmyadmin:latest
```

Then you can create db1.php , db2.php , or any file name you want, and store them in the `conf.d` directory on the host.

**Comment by williamdes** [comment]
> I use google translate to translate my thoughts into English, what do you think?

Okay, It's doing well. Do you speak other languages ?

----

You can change the line from
```diff
- /some/local/directory/conf.d:/etc/phpmyadmin/conf.d
+ /some/local/directory/conf.d:/etc/phpmyadmin/conf.d:ro
```
To make the mount read-only.


> you can also consider storing your custom configuration files in /etc/phpmyadmin/conf.d

you can also consider storing your custom configuration files in the folder `/etc/phpmyadmin/conf.d`

> which is very suitable for managing multiple MySQL Container

which is very suitable for managing multiple phpMyAdmin configuration files for different hosts


> Then you can create db1.php , db2.php , or any file name you want, and store them in the conf.d directory on the host.

Then you can create `server-1.php` , `server-2.php` , or any file name you want, and store them in the conf.d directory mounted on the host.

**Comment by pigochu** [comment]
> Okay, It's doing well. Do you speak other languages ?
Sorry, I only speak Chinese and poor English

**Comment by williamdes** [comment]
> Sorry, I only speak Chinese and poor English

Okay, if you are interested (no obligation at all) you can help with translations in Chinese:
- https://hosted.weblate.org/projects/phpmyadmin/5-2/zh_Hant/ (Traditional)
- https://hosted.weblate.org/projects/phpmyadmin/5-2/zh_Hans/ (Simplified)

**Comment by williamdes** [comment]
let me know when it's okay for the README. I can improve it while I merge your work

File: README.md [hunk 2]
```diff
@@ -136,6 +136,17 @@ See the following links for config file information:
 * https://docs.phpmyadmin.net/en/latest/config.html#config
 * https://docs.phpmyadmin.net/en/latest/setup.html
 
+## Adding Custom Configuration in /etc/phpmyadmin/conf.d
+
+you can also consider storing your custom configuration files in the folder /etc/phpmyadmin/conf.d, which is very suitable for managing multiple phpMyAdmin configuration files for different hosts,Then you can create server-1.php , server-2.php , or any file name you want, and store them in the conf.d directory mounted on the host.
```
Comment (williamdes) [inline README.md:141]:
```suggestion
you can also consider storing your custom configuration files in the folder `/etc/phpmyadmin/conf.d`, which is very suitable for managing multiple phpMyAdmin configuration files for different hosts.
Then you can create `server-1.php`, `server-2.php`, or any file name you want, and store them in the `conf.d` directory mounted on the host.
```

**Review by williamdes — APPROVED** [review]
One line you could add is about permissions, the folder and files should be readable from the phpMyAdmin container. Or it will not work at all.

**Comment by pigochu** [comment]
> One line you could add is about permissions, the folder and files should be readable from the phpMyAdmin container. Or it will not work at all.

I have a question, is it necessary to use `ro`?
In fact, I have another project that wants to automatically discovers  mysql service  and automatically modify or add the files in conf.d . So I need write permission.

**Comment by williamdes** [comment]
> I have a question, is it necessary to use `ro`?
> In fact, I have another project that wants to automatically discovers mysql service and automatically modify or add the files in conf.d . So I need write permission.

From a security point of view, it's better to say to the user we do not need write access. If you need write access, feel free to use `rw` or nothing :)

**Pull request merged**